src/utils/allocators.zig

```diff
@@ -136,38 +136,78 @@ pub fn RecycleFBA(config: struct {
     };
 }
 
+test "recycle allocator" {
```

Comment:
can we keep all tests on the bottom of the file

Reply:
I almost left the same comment on this pr. But it grew on me a little. There is something nice about the test being immediately adjacent to the struct being tested. It's a little unexpected though since the rest of the project does not do this.

Reply:
it has not grown on me hahaha

Reply:
e5e987370512681ddbfe300f4d18a9edcd52f626